### ISSUE TITLE
Example of how to use

### ISSUE THREAD

**Issue by johntdyer** [logstash-plugins/logstash-input-jms#7]
Anyone have an example of hiw to use this ?  Its not very clear to me from the source and its not showing up on the elasti.co site

**Comment by AlphaCluster** [comment]
The documentation in the main plugin file cover most of the parameters.
https://github.com/logstash-plugins/logstash-input-jms/blob/master/lib/logstash/inputs/jms.rb
In general you have two ways to configure it you can configure the jms options for the connection in the properties for the input or you can configure the jms end of it via a yml file which you point to via the properties. There is an example configuration in the jms.rb file which shows how that might be configured.

If you want to use the yml method where you configure jms via yml you can see an example yml file in the specs folder. If you want to see some more examples of this checkout jruby-jms
https://github.com/logstash-plugins/logstash-input-jms/blob/master/spec/inputs/jms.yml

Hopefully that helps!

**Comment by JiriKadlcik** [comment]
The source is indeed very comprehensive documentation itself, however there should be a page describing the configuration options and examples readable to humans as well.

**Comment by ranierimazili** [comment]
I really don't know how to use it and unfortunatelly, I need it.
Can you please post an example of how to configure your plugin?

I mean, a configuration like this one would be awesome: https://www.elastic.co/guide/en/logstash/2.4/plugins-inputs-jmx.html

I don't know where to add the configuration. Trying to use yml, I don't know why you choose HornetQ as context, mine is WebSphere, should I define it as websphere: instead of hornetq:?

Too much doubts without answers :) Sorry and thanks for you attention

**Comment by AgentW4C** [comment]
@ranierimazili I'm preparing more compherensive docs, but will share part of a sample configuration for WebshpereMQ here. My logstash configuration includes:

```
input {
  jms {
    include_header => true
    include_properties => true
    include_body => true
    use_jms_timestamp => true
    interval => 10
    destination => "TEST01"
    yaml_file => "/etc/logstash/logstash-input-jms/dcg1-mq.yml"
    yaml_section => "dcg-dev"
    codec => "json"
  }
}
```

and the referenced yaml file `/etc/logstash/logstash-input-jms/dcg1-mq.yml` contains:

```
dev:
  :factory: com.ibm.mq.jms.MQQueueConnectionFactory
  :queue_manager: MQDEV01
  :host_name: mq-dev.example.com
  :port: 1414
  :channel: LOGSTASH
  :transport_type: 1
  :require_jars:
    - /opt/mqm/java/lib/com.ibm.dhbcore.jar
    - /opt/mqm/java/lib/com.ibm.mq.axis2.jar
    - /opt/mqm/java/lib/com.ibm.mq.commonservices.jar
    - /opt/mqm/java/lib/com.ibm.mq.headers.jar
    - /opt/mqm/java/lib/com.ibm.mq.jar
    - /opt/mqm/java/lib/com.ibm.mq.jmqi.jar
    - /opt/mqm/java/lib/com.ibm.mq.jms.Nojndi.jar
    - /opt/mqm/java/lib/com.ibm.mq.pcf.jar
    - /opt/mqm/java/lib/com.ibm.mq.soap.jar
    - /opt/mqm/java/lib/com.ibm.mq.tools.ras.jar
    - /opt/mqm/java/lib/com.ibm.mqjms.jar
    - /opt/mqm/java/lib/javax.jms.jar
    - /opt/mqm/java/lib/javax.resource.jar
    - /opt/mqm/java/lib/javax.transaction.jar
```

I think there might be less jars referenced in `:require_jars:` (will be part of further tunning) but I simply added everything what I had to make it running first.

**Comment by robbavey** [comment]
This has been fixed by #35
